tune queue values for vnic adapter

vaishnavibhat opened the pull request:
- tune vnic adapter queue to max supported values
- Use dmesg level=4 for vnic related tests

vaishnavibhat commented:
avocado run network_virtualization.py -m network_virtualization.py.data/network_virtualization.yaml --max-parallel-tasks=1
JOB ID     : db2c1a5f3ec8f44d41a779ffda3d9dcb92939d3d
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-07-30T06.25-db2c1a5/job.log
 (01/12) network_virtualization.py:NetworkVirtualization.test_add;run-6ca3: STARTED
 (01/12) network_virtualization.py:NetworkVirtualization.test_add;run-6ca3: PASS (28.89 s)
 (02/12) network_virtualization.py:NetworkVirtualization.test_backingdevadd;run-6ca3: STARTED
 (02/12) network_virtualization.py:NetworkVirtualization.test_backingdevadd;run-6ca3: PASS (22.94 s)
 (03/12) network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-6ca3: STARTED
 (03/12) network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-6ca3: FAIL: test failed,check dmesg log in debug log (20.52 s)
 (04/12) network_virtualization.py:NetworkVirtualization.test_hmcfailover;run-6ca3: STARTED
 (04/12) network_virtualization.py:NetworkVirtualization.test_hmcfailover;run-6ca3: PASS (83.28 s)
 (05/12) network_virtualization.py:NetworkVirtualization.test_clientfailover;run-6ca3: STARTED
 (05/12) network_virtualization.py:NetworkVirtualization.test_clientfailover;run-6ca3: PASS (136.29 s)
 (06/12) network_virtualization.py:NetworkVirtualization.test_vnic_auto_failover;run-6ca3: STARTED
 (06/12) network_virtualization.py:NetworkVirtualization.test_vnic_auto_failover;run-6ca3: PASS (94.67 s)
 (07/12) network_virtualization.py:NetworkVirtualization.test_rmdev_viosfailover;run-6ca3: STARTED
 (07/12) network_virtualization.py:NetworkVirtualization.test_rmdev_viosfailover;run-6ca3: PASS (84.41 s)
 (08/12) network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-6ca3: STARTED
 (08/12) network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-6ca3: PASS (48.98 s)
 (09/12) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-6ca3: STARTED
 (09/12) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-6ca3: PASS (66.50 s)
 (10/12) network_virtualization.py:NetworkVirtualization.test_vnic_eeh;run-6ca3: STARTED
 (10/12) network_virtualization.py:NetworkVirtualization.test_vnic_eeh;run-6ca3: PASS (83.53 s)
 (11/12) network_virtualization.py:NetworkVirtualization.test_backingdevremove;run-6ca3: STARTED
 (11/12) network_virtualization.py:NetworkVirtualization.test_backingdevremove;run-6ca3: PASS (13.62 s)
 (12/12) network_virtualization.py:NetworkVirtualization.test_remove;run-6ca3: STARTED
 (12/12) network_virtualization.py:NetworkVirtualization.test_remove;run-6ca3: PASS (15.01 s)
RESULTS    : PASS 11 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-07-30T06.25-db2c1a5/results.html
JOB TIME   : 740.79 s

Test summary:
03-network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-6ca3: FAIL


[job.log](https://github.com/user-attachments/files/16426401/job.log)

vaishnavibhat commented:
avocado run network_virtualization.py -m network_virtualization.yaml --max-parallel-tasks=1 
JOB ID     : ab5f5a979c65a381eaa533707f66f3e459198eb8
JOB LOG    : /root/vnic/avocado-fvt-wrapper/results/job-2024-07-30T05.56-ab5f5a9/job.log
 (01/12) network_virtualization.py:NetworkVirtualization.test_add;run-c923: STARTED
 (01/12) network_virtualization.py:NetworkVirtualization.test_add;run-c923: PASS (22.48 s)
 (02/12) network_virtualization.py:NetworkVirtualization.test_backingdevadd;run-c923: STARTED
 (02/12) network_virtualization.py:NetworkVirtualization.test_backingdevadd;run-c923: PASS (18.90 s)
 (03/12) network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-c923: STARTED
 (03/12) network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-c923: FAIL: test failed,check dmesg log in debug log (17.69 s)
 (04/12) network_virtualization.py:NetworkVirtualization.test_hmcfailover;run-c923: STARTED
 (04/12) network_virtualization.py:NetworkVirtualization.test_hmcfailover;run-c923: PASS (79.22 s)
 (05/12) network_virtualization.py:NetworkVirtualization.test_clientfailover;run-c923: STARTED
 (05/12) network_virtualization.py:NetworkVirtualization.test_clientfailover;run-c923: PASS (135.71 s)
 (06/12) network_virtualization.py:NetworkVirtualization.test_vnic_auto_failover;run-c923: STARTED
 (06/12) network_virtualization.py:NetworkVirtualization.test_vnic_auto_failover;run-c923: PASS (86.03 s)
 (07/12) network_virtualization.py:NetworkVirtualization.test_rmdev_viosfailover;run-c923: STARTED
 (07/12) network_virtualization.py:NetworkVirtualization.test_rmdev_viosfailover;run-c923: PASS (85.99 s)
 (08/12) network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-c923: STARTED
 (08/12) network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-c923: PASS (78.92 s)
 (09/12) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-c923: STARTED
 (09/12) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-c923: PASS (72.63 s)
 (10/12) network_virtualization.py:NetworkVirtualization.test_vnic_eeh;run-c923: STARTED
 (10/12) network_virtualization.py:NetworkVirtualization.test_vnic_eeh;run-c923: PASS (84.58 s)
 (11/12) network_virtualization.py:NetworkVirtualization.test_backingdevremove;run-c923: STARTED
 (11/12) network_virtualization.py:NetworkVirtualization.test_backingdevremove;run-c923: PASS (11.21 s)
 (12/12) network_virtualization.py:NetworkVirtualization.test_remove;run-c923: STARTED
 (12/12) network_virtualization.py:NetworkVirtualization.test_remove;run-c923: PASS (14.05 s)
RESULTS    : PASS 11 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/vnic/avocado-fvt-wrapper/results/job-2024-07-30T05.56-ab5f5a9/results.html
JOB TIME   : 750.60 s

Test summary:
03-network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-c923: FAIL

[job.log](https://github.com/user-attachments/files/16426747/job.log)

vaishnavibhat commented on io/net/virt-net/network_virtualization.py:
```diff
@@ -257,12 +256,12 @@ def tune_rxtx_queue(self):
         peer_interface = self.remotehost.get_interface_by_ipaddr(self.peer_ip[0]).name
-        cmd = "ethtool -L %s rx 16 tx 16" % peer_interface
-        output = self.session.cmd(cmd)
+        cmd = "ethtool -L %s rx 10 tx 10" % peer_interface
```
> @vaishnavibhat So the max queue support is 10 now? reduced from 16 to 10.

vnic supports queue upto 10.  Current hardware settings changes to 10 inspite of giving max (16) value. 

#ethtool -L env9 rx 16 tx 16
#ethtool -l env9
Channel parameters for env9:
Pre-set maximums:
RX:		16
TX:		16
Other:		n/a
Combined:	n/a
Current hardware settings:
RX:		10
TX:		10
Other:		n/a
Combined:	n/a

abdhaleegit commented on io/net/virt-net/network_virtualization.py:
```diff
@@ -257,12 +256,12 @@ def tune_rxtx_queue(self):
             self.cancel("Unable to tune RX and TX queue in peer")
         device = self.find_device(self.mac_id[0])
-        cmd = "ethtool -L %s rx 16 tx 16" % device
+        cmd = "ethtool -L %s rx 10 tx 10" % device
```
make it yaml parameter max_transmit: 10 defaults.

abdhaleegit reviewed (changes requested):
please test once with few other tests as cfg bucket,

vaishnavibhat commented:
After addressing the review comments :

avocado run network_virtualization.py -m network_virtualization.py.data/network_virtualization.yaml --max-parallel-tasks=1 
JOB ID     : 9ac6cdb6b092d1c20f145b8da8802e11a4a81e94
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-07-31T05.54-9ac6cdb/job.log
 (01/12) network_virtualization.py:NetworkVirtualization.test_add;run-64c9: STARTED
 (01/12) network_virtualization.py:NetworkVirtualization.test_add;run-64c9: PASS (33.87 s)
 (02/12) network_virtualization.py:NetworkVirtualization.test_backingdevadd;run-64c9: STARTED
 (02/12) network_virtualization.py:NetworkVirtualization.test_backingdevadd;run-64c9: PASS (35.71 s)
 (03/12) network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-64c9: STARTED
 (03/12) network_virtualization.py:NetworkVirtualization.test_disable_enable_dev;run-64c9: PASS (30.07 s)
 (04/12) network_virtualization.py:NetworkVirtualization.test_hmcfailover;run-64c9: STARTED
 (04/12) network_virtualization.py:NetworkVirtualization.test_hmcfailover;run-64c9: PASS (91.40 s)
 (05/12) network_virtualization.py:NetworkVirtualization.test_clientfailover;run-64c9: STARTED
 (05/12) network_virtualization.py:NetworkVirtualization.test_clientfailover;run-64c9: PASS (144.58 s)
 (06/12) network_virtualization.py:NetworkVirtualization.test_vnic_auto_failover;run-64c9: STARTED
 (06/12) network_virtualization.py:NetworkVirtualization.test_vnic_auto_failover;run-64c9: PASS (104.31 s)
 (07/12) network_virtualization.py:NetworkVirtualization.test_rmdev_viosfailover;run-64c9: STARTED
 (07/12) network_virtualization.py:NetworkVirtualization.test_rmdev_viosfailover;run-64c9: PASS (91.54 s)
 (08/12) network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-64c9: STARTED
 (08/12) network_virtualization.py:NetworkVirtualization.test_vnic_dlpar;run-64c9: PASS (57.11 s)
 (09/12) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-64c9: STARTED
 (09/12) network_virtualization.py:NetworkVirtualization.test_vnic_hmc_dlpar;run-64c9: PASS (75.11 s)
 (10/12) network_virtualization.py:NetworkVirtualization.test_vnic_eeh;run-64c9: STARTED
 (10/12) network_virtualization.py:NetworkVirtualization.test_vnic_eeh;run-64c9: PASS (83.68 s)
 (11/12) network_virtualization.py:NetworkVirtualization.test_backingdevremove;run-64c9: STARTED
 (11/12) network_virtualization.py:NetworkVirtualization.test_backingdevremove;run-64c9: PASS (22.01 s)
 (12/12) network_virtualization.py:NetworkVirtualization.test_remove;run-64c9: STARTED
 (12/12) network_virtualization.py:NetworkVirtualization.test_remove;run-64c9: PASS (15.29 s)
RESULTS    : PASS 12 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-07-31T05.54-9ac6cdb/results.html
JOB TIME   : 826.50 s

vaishnavibhat commented:
[job.log](https://github.com/user-attachments/files/16439862/job.log)

Naresh-ibm reviewed (approved):
@vaishnavibhat Thanks for patch
LGTM
